server/controllers/cohort.js

```diff
@@ -488,15 +488,64 @@ const addMember = (req, resp) => {
     });
 };
 
+const leaveCohort = (req, resp) => {
+  const { id: cohortId } = req.params;
+  const { userId } = req.body;
+  const sanitizedUserId = sanitize(userId);
+  const sanitizedCohortId = sanitize(cohortId);
+
+  Cohort.findOne({
+    _id: sanitizedCohortId,
+    memberIds: { $in: [sanitizedUserId] }
+  })
+    .exec()
+    .then(doc => {
+      if (!doc) {
+        throw new BadRequestException();
+      }
+
+      const { memberIds, ownerIds } = doc;
+      const isOwner = checkIfArrayContainsUserId(ownerIds, userId);
+      const isMember = checkIfArrayContainsUserId(memberIds, userId);
+
+      if (ownerIds.length === 1 && isOwner) {
```

Comment:
1. first check `isOwner` - if not then the length is not important for us.
2. merge two if into one
```
if (isOwner) {
    if (ownerIds.length === 1) {
        throw new ...
    }

    ownerIds.splice...
}